presto-main/src/main/java/com/facebook/presto/operator/project/PageProcessor.java

```diff
@@ -103,6 +105,17 @@ public PageProcessorOutput process(ConnectorSession session, Page page)
         return projections;
     }
 
+    private static long calculatePageRetainedSizeWithoutLoading(Page page)
+    {
+        long retainedSizeInBytes = 0;
+        for (Block block : page.getBlocks()) {
+            if (!(block instanceof LazyBlock)) {
+                retainedSizeInBytes += block.getRetainedSizeInBytes();
```

Comment:
:)  That what I did the first time.

There are places that assume retained size is constant for the life of the block, and cache the results.  Currently, `PageProcessor` is one of a few (only?) place that sees an unloaded `LazyBlock`, so for now, I think it is safer to have have custom handling here.  In the long run, if we use `LazyBlock` in more places, we will have to rethink the way size calculations work.